services/builder/configProcessor.go

```diff
@@ -190,7 +190,7 @@ func configProcessor(c pb.ContextBoxServiceClient, wg *sync.WaitGroup) error {
 					return err
 				}
 				logger.Info().Msgf("Deleting nodes from cluster")
-				cluster, err := deleteNodes(clusterView.CurrentClusters[clusterName], diff.ToDelete)
+				cluster, err := deleteNodes(clusterView.CurrentClusters[clusterName], clusterView.DesiredClusters[clusterName], diff.ToDelete)
```

Comment:
Because for static node pools, the deletion of a node is done by manually removing individual nodes from the manifest. Thus counts to delete alone are not cutting it, as we need to know which exact node has to be deleted (compared to the dynamic node pool, where we can delete any, as long as the count matches). 

Thus, by checking which static nodes are in the current state and which are not in the desired state, we identify the node which needs to be deleted.